runtime/outer-PEC.js

```diff
@@ -101,11 +108,22 @@ class OuterPEC extends PEC {
           }
           handle.mapToStorage(fromHandle);
         }
+        let normalized = false;
         if (missingHandles.length > 0) {
-          error = `Recipe couldn't load due to missing handles [recipe=${recipe0}, missingHandles=${missingHandles.join('\n')}].`;
-        } else {
+          // Attempt to resolve missing handles with the strategizer.
+          const resolvedRecipes = await OuterPEC._applyResolveRecipe(this._arc, recipe0);
+          if (resolvedRecipes.length == 0) {
+            error = `Recipe couldn't load due to missing handles [recipe=${recipe0}, missingHandles=${missingHandles.join('\n')}].`;
+          } else {
+            assert(resolvedRecipes.length == 1, `Strategizer shouldn't produce more than one recipe with a single input recipe.`);
+            recipe0 = resolvedRecipes[0];
+            // The strategizer normalizes the recipe as part of its flow.
+            normalized = true;
+          }
+        }
+        if (!error) {
           let options = {errors: new Map()};
-          if (recipe0.normalize(options)) {
+          if (normalized || recipe0.normalize(options)) {
```

Comment:
maybe instead you could do
```
if (missingHandles.length == 0) {
  recipe0.normalize(options);
} else {
  // do all the resolution stuff here.
}
```
then you don't need the `normalize` variable and the code is more readable.

Reply:
Cleaned up along these lines.